openassessment/workflow/models.py

```diff
@@ -357,7 +360,9 @@ def update_from_assessments(self, assessment_requirements):
             score = self.get_score(assessment_requirements, step_for_name)
             # If we found a score, then we're done
             if score is not None:
-                self.set_score(score)
+                # Only set the score if it's not a staff score, in which case it will have already been set above
+                if score.get("staff_id") is None:
```

Comment:
This is to prevent setting a staff score twice, since it was already checked/set above, before the `if status is done` line.